quantmsrescore/ms2_model_manager.py

```
@@ -33,20 +70,67 @@ def __init__(self,
+
+        Uses streaming download to avoid loading entire file into memory,
+        and a longer timeout (300s) for large files on slow connections.
+        """
+        url = self.model_url
+        parsed = urllib.parse.urlparse(url)
+        if parsed.scheme not in ("http", "https"):
+            raise ValueError(f"Disallowed URL scheme: {parsed.scheme}")
+
+        if os.path.exists(model_zip_file_path):
+            if not overwrite:
+                raise FileExistsError(f"Model file already exists: {model_zip_file_path}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)